app/models/nic/base.rb

```diff
@@ -202,6 +204,19 @@ def sync_name
       end
     end
 
+    def validate_host_taxonomy
+      return true if self.subnet.nil?
```

Comment:
Maybe something like the following would not require maintenance of 'locations' 'organizations', etc..

```ruby
Taxonomy.enabled_taxonomies.each do |taxonomy|
  next if subnet.send(:taxonomy).empty? || self.send(:"#{taxonomy.singularize}").empty?
  errors.add(:subnet, _("is not defined for host's #{taxonomy.singularize}.")) unless subnet.send(:taxonomy).include? self.send(:"#{taxonomy.singularize}")
end
```

This change plus making the validation conditional would avoid doing a lot of the taxonomy checks which could be (it's previously been) a source of bugs.